ide/che-core-ide-generators/src/main/java/org/eclipse/che/util/GwtXmlGenerator.java

```diff
@@ -0,0 +1,238 @@
+        return values != null ? values.iterator().next() : defaultValue;
+    }
+
+
+    /**
+     * Class provides functionality of searching XXX.gwt.xml files in class path
+     */
+    public static class GwtXmlModuleSearcher {
+        private final Set<String> excludePackages;
+        private final Set<String> includePackages;
+        private final Set<URL>    urls;
+
+        public GwtXmlModuleSearcher(Set<String> excludePackages, Set<String> includePackages, Set<URL> urls) {
+            this.excludePackages = excludePackages;
+            this.includePackages = includePackages;
+            this.urls = urls;
+        }
+
+        /**
+         * Searches XXX.gwt.xml files in class path
+         *
+
+         * @return - set of XXX.gwt.xml files found in class path
+         */
+        public Set<String> getGwtModulesFromClassPath() {
+            ConfigurationBuilder configurationBuilder = getConfigurationBuilder();
+            if (urls != null && urls.size() > 0) {
+                configurationBuilder.addUrls(urls);
+            }
+            FilterBuilder filterBuilder = new FilterBuilder();
+            for (String includePackage : includePackages) {
+                filterBuilder.includePackage(includePackage);
+            }
+            for (String excludePackage : excludePackages) {
+                filterBuilder.excludePackage(excludePackage);
+            }
+
+
+            configurationBuilder.setScanners(new ResourcesScanner()).filterInputsBy(filterBuilder);
+
+            Reflections reflection = new Reflections(configurationBuilder);
+            return new TreeSet<>(reflection.getResources(name -> name.endsWith(".gwt.xml")));
+
+        }
+    }
+
+    public static class GwtXmlGeneratorConfig {
+        private final Set<String> gwtXmlModules;
+
+        private final File generationRoot;
+
+        private final String gwtFileName;
+
+        private final String entryPoint;
+
+        private final String stylesheet;
+
+        private final boolean isLoggingEnabled;
+
+        public GwtXmlGeneratorConfig(Set<String> gwtXmlModules,
+                                     File generationRoot,
+                                     String gwtFileName,
+                                     String entryPoint,
+                                     String stylesheet,
+                                     boolean isLoggingEnabled) {
+            this.gwtXmlModules = gwtXmlModules;
+            this.generationRoot = generationRoot;
+            this.gwtFileName = gwtFileName;
+            this.entryPoint = entryPoint;
+            this.stylesheet = stylesheet;
+            this.isLoggingEnabled = isLoggingEnabled;
+        }
+
+        public GwtXmlGeneratorConfig(Set<String> gwtXmlModules,
+                                     File generationRoot) {
+            this(gwtXmlModules, generationRoot, DEFAULT_GWT_XML_PATH, DEFAULT_GWT_ETNRY_POINT, DEFAULT_STYLE_SHEET, false);
```

Comment:
Would it better to add constant for `isLoggingEnabled` and reuse it here and [there](https://github.com/eclipse/che/pull/4298/files#diff-da04613073f2bdb0b2608042c5d49967R116)